indy-didresolver/src/resolver.rs

```diff
@@ -0,0 +1,122 @@
+use std::str::FromStr;
+
+use futures_executor::block_on;
+use indy_vdr::utils::Qualifiable;
+use serde_json::Value;
+
+use super::did::{did_parse, LedgerObject};
+use super::did_document::DidDocument;
+use super::error::{DidIndyError};
+use super::responses::{GetNymResult};
+
+use indy_vdr::common::error::VdrResult;
+use indy_vdr::ledger::constants::{GET_NYM};
+use indy_vdr::ledger::identifiers::{CredentialDefinitionId, SchemaId};
+use indy_vdr::pool::helpers::perform_ledger_request;
+use indy_vdr::pool::{Pool, PreparedRequest, RequestResult, TimingResult};
+use indy_vdr::utils::did::DidValue;
+
+use ssi::did;
+
+pub struct Resolver<T: Pool> {
+    pool: T,
+}
+
+impl<T: Pool> Resolver<T> {
+    pub fn new(pool: T) -> Resolver<T> {
+        Resolver { pool }
+    }
+
+    pub fn resolve(&self, did: &str) -> Result<String, DidIndyError> {
+        let did_url = did::DIDURL::try_from(String::from(did)).expect("Could not parse DID URL");
+
+        let did = did_url.did;
+        let did = match did_parse(did.as_str()) {
+            Ok(did) => did,
+            Err(DidIndyError) => {
+                return Err(DidIndyError);
+            }
+        };
+
+        // The path variable identifies the requested ledger object
+        // If there is no path, then we return a DID document
+        let path = if did_url.path_abempty != "" {
+            Some(did_url.path_abempty)
+        } else {
+            None
+        };
+
+        println!("{:?}", path);
+
+        let builder = self.pool.get_request_builder();
+        let did_value = DidValue::new(&did.id, Option::None);
+
+        let request = match path {
```

Comment:
maybe separate this block to build_request function as it will grow even larger with revoc_reg_def/entry?

Reply:
Yep, makes sense.

Reply:
I introduced build_request but needs another signature later on since we'll need the DID URL fragment as well. We;; probably feed the whole DID URL to this function.